src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,235 @@
+                    { SyntaxKind.LessThanOrEqualExpression, (SyntaxKind.GreaterThanExpression, SyntaxKind.GreaterThanToken) },
+                    { SyntaxKind.GreaterThanExpression, (SyntaxKind.LessThanOrEqualExpression, SyntaxKind.LessThanEqualsToken) },
+                    { SyntaxKind.GreaterThanOrEqualExpression, (SyntaxKind.LessThanExpression, SyntaxKind.LessThanToken) },
+                };
+
+        protected override SyntaxNode GetIfStatement(TextSpan textSpan, SyntaxToken token, CancellationToken cancellationToken)
+        {
+            var ifStatement = token.GetAncestor<IfStatementSyntax>();
+            if (ifStatement == null || ifStatement.Else == null)
+            {
+                return null;
+            }
+
+            var span = TextSpan.FromBounds(ifStatement.GetFirstToken().Span.Start, ifStatement.CloseParenToken.Span.End);
+            if (!span.IntersectsWith(textSpan))
+            {
+                return null;
+            }
+
+            return ifStatement;
+        }
+
+        protected override SyntaxNode GetRootWithInvertIfStatement(Workspace workspace, SemanticModel model, SyntaxNode ifStatement, CancellationToken cancellationToken)
+        {
+        var ifNode = (IfStatementSyntax)ifStatement;
+
+        // For single line statment, we swap the TrailingTrivia to preserve the single line
+        StatementSyntax newIfNodeStatement = null;
+        ElseClauseSyntax newElseStatement = null;
+        var newCondition = Negate(ifNode.Condition, model, cancellationToken);
+
+        if (ifNode.Statement.GetTrailingTrivia().Any(trivia => trivia.Kind() == SyntaxKind.EndOfLineTrivia))
+        {
+            // multiline if
+            newIfNodeStatement = ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement  // if it's an else-if
+            ? SyntaxFactory.Block(ifNode.Else.Statement)
+            : ifNode.Else.Statement;
+            newElseStatement = ifNode.Else.WithStatement(ifNode.Statement);
+            ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken))
+                .WithStatement(ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block ? SyntaxFactory.Block(newIfNodeStatement) : newIfNodeStatement)
+                .WithElse(newElseStatement)
+                .WithAdditionalAnnotations(Formatter.Annotation);
+        }
+        else
+        {
+            // singleline if
+            var elseTrailingTrivia = ifNode.Else.GetTrailingTrivia();
+            var ifTrailingTrivia = ifNode.Statement.GetTrailingTrivia();
+
+            newIfNodeStatement = ifNode.Else.Statement.WithTrailingTrivia(ifTrailingTrivia);
+            newElseStatement = ifNode.Else.WithStatement(ifNode.Statement).WithTrailingTrivia(elseTrailingTrivia);
+            ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken))
```

Comment:
Can we share some code between if and else parts by moving this statement outside the if-else statement?
```
   ifNode = ifNode.WithCondition(Negate(ifNode.Condition, model, cancellationToken)) 
            .WithStatement(ifNode.Else.Statement.Kind() == SyntaxKind.IfStatement && newIfNodeStatement.Kind() != SyntaxKind.Block ?  SyntaxFactory.Block(newIfNodeStatement) : newIfNodeStatement)
            .WithElse(newElseStatement); 
```